src/oracle/types/mod.rs

```diff
@@ -15,77 +15,62 @@ pub type ToSqlResult = FromSqlResult<IsNull>;
-        size: usize,
-        scale: Option<i8>,
-        precision: i16,
-    },
-    Float {
-        raw_type: u32,
-        size: usize,
-    },
-    Date {
-        raw_type: u32,
-        size: usize,
-    },
-    Blob {
-        raw_type: u32,
-    },
-    Text {
-        raw_type: u32,
-    },
-}
-
-// #[allow(dead_code)]
-// #[repr(u32)]
-// #[derive(Clone, Copy, Debug, PartialEq, Hash, Eq)]
-// pub enum OCIDataType {
-//     Char = ffi::SQLT_CHR,
-//     Date = ffi::SQLT_DATE,
-//     Time = ffi::SQLT_TIME,
-//     Timestamp = ffi::SQLT_TIMESTAMP,
-//     TimestampWithTz = ffi::SQLT_TIMESTAMP_TZ,
-//     TimestampWithLocalTz = ffi::SQLT_TIMESTAMP_LTZ,
-//     IntervalYearToMonth = ffi::SQLT_INTERVAL_YM,
-//     IntervalDayToSecond = ffi::SQLT_INTERVAL_DS,
-//     Clob = ffi::SQLT_CLOB,
-//     Blob = ffi::SQLT_BLOB,
-//     Int = ffi::SQLT_INT,
-//     Uint = ffi::SQLT_UIN,
-//     Float = ffi::SQLT_FLT,
-//     PackedDecimalNumber = ffi::SQLT_PDN,
-//     Binary = ffi::SQLT_BIN,
-//     Numeric = ffi::SQLT_NUM,
-//     NamedObject = ffi::SQLT_NTY,
-//     Ref = ffi::SQLT_REF,
-//     OCIString = ffi::SQLT_VST,
-//     NumericWithLength = ffi::SQLT_VNU,
-//     BFloat = ffi::SQLT_BFLOAT,
-//     BDouble = ffi::SQLT_BDOUBLE,
-//     IBFloat = ffi::SQLT_IBFLOAT,
-//     IBDouble = ffi::SQLT_IBDOUBLE,
-//     String = ffi::SQLT_STR,
-//     AnsiChar = ffi::SQLT_AFC,
-//     InternDate = ffi::SQLT_DAT,
-// }
+    Bool,
+    SmallInt,
+    Integer,
+    BigInt,
+    Float,
+    Double,
+    Text,
+    Binary,
+    Date,
+    Time,
+    Timestamp,
+}
 
 impl OciDataType {
     pub fn is_text(&self) -> bool {
         match *self {
-            OciDataType::Text { .. } => true,
+            OciDataType::Text => true,
             _ => false,
         }
     }
 
-    pub fn as_raw(&self) -> u32 {
+    pub fn bind_type(&self) -> u32 {
+        use self::OciDataType::*;
+        match *self {
+            Bool => ffi::SQLT_INT,
+            SmallInt => ffi::SQLT_INT,
+            Integer => ffi::SQLT_INT,
+            BigInt => ffi::SQLT_INT,
+            Float => ffi::SQLT_IBFLOAT,
+            Double => ffi::SQLT_IBDOUBLE,
+            Text => ffi::SQLT_CHR,
+            Binary => ffi::SQLT_BIN,
+            Date | Time | Timestamp => ffi::SQLT_DAT,
+        }
+    }
+
+    pub fn define_type(&self) -> u32 {
+        use self::OciDataType::*;
+        match *self {
+            Text => ffi::SQLT_STR,
+            _ => self.bind_type(),
+        }
+    }
+
+    pub fn byte_size(&self) -> usize {
         use self::OciDataType::*;
         match *self {
-            Float { raw_type, .. }
-            | Text { raw_type, .. }
-            | Number { raw_type, .. }
-            | Date { raw_type, .. }
-            | Blob { raw_type, .. } => raw_type,
+            Bool => 2,
+            SmallInt => 2,
+            Integer => 4,
+            BigInt => 8,
+            Float => 4,
+            Double => 8,
+            Text => 2_000_000,
```

Comment:
Is that a max size for a particular container or the size always used? If the later, I would rather not do this, otherwise we allocate 2GB per `text` column, I don't think we want that?

Reply:
It's 2MB, which should be fine for now. We will try to fix this later, but that requires some more work in the defining and returning case.